### PR TITLE
Use instance keys

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

**Comment by mattcollier** [comment]
@gannan08 @davidlehn this PR was started many moons ago so will not show up in any recent PR list, but needs review please.

File: instances.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    throw e;
+  }
+}
+
+export async function _getInstanceIds({id}) {
+  const zcaps = await getCapabilities({id});
+  const ids = [];
+  zcaps.forEach(({referenceId}) => {
+    // assumes instance ids are 36 characters and prepended on referenceId
+    const instanceId = referenceId.slice(0, 36);
```
Comment (davidlehn) [inline instances.js]: This looks fragile.

Comment (gannan08) [reply instances.js]: Agreed. What is your recommendation for a proposed change? Currently, the Issuer Instance's ID is a uuid. That uuid is prepended to other data to create the reference id. I could not think of another way of getting the information I need.

**Comment by dlongley** [comment]
As a basis for forward progress, I'd like to get this merged to master and then pull in the other PRs that targeted this branch.

cc: @mattcollier, @gannan08

**Pull request merged**